src/Raven.Server/Rachis/Leader.cs

```diff
@@ -183,7 +184,7 @@ private void RefreshAmbassadors(ClusterTopology clusterTopology)
                     }
 
                     var ambasaddor = new FollowerAmbassador(_engine, this, _voterResponded, voter.Key, voter.Value,
-                        _engine.ClusterCertificate);
+                        _engine.ClusterCertificate, connections?[voter.Key]);
```

Comment:
`KeyNotFoundException` can be raised here, make sure that you use `TryGetValue`, instead.